scripts/update.sh

```diff
@@ -255,6 +255,17 @@ then
 	fi
 fi
 
+### delicate: updating the update script itself ###
+# ensure that update.sh does not mutate itself while running
+# emergency rescue: on erroneous EXIT recover update.sh from a tmp-file
+# (only replaces the file if a tmp-file exists)
+recover_update_sh () {
+	cp update_tmp.sh update.sh
+}
+trap recover_update_sh EXIT
+# rename the current script to ensure that we do not override what we are executing
+mv $0 update_tmp.sh && cp update_tmp.sh update.sh
```

Comment:
I don't know precisely how /bin/sh, which this script is configured to run as, behaves compared to /bin/bash, but with bash this would have to be:
```mv "$0" ...```
See http://mywiki.wooledge.org/BashPitfalls#cp_.24file_.24target

Further, independent of the shell as that is a feature of mv, this actually needs to be:
```mv -- ...```
See http://mywiki.wooledge.org/BashPitfalls#Filenames_with_leading_dashes

Reply:
ah, yes, thank you!